package/scripts/postinstall

```diff
@@ -21,35 +21,17 @@ fi
 # add Git to path
 export PATH="/Library/Developer/CommandLineTools/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:${PATH}"
 
-# helpers for setting/unsetting Git's safe directory setting
-set_git_safe_directory() {
-  if git config --global --get-all safe.directory | grep -q "${1}"
-  then
-    return
-  fi
-  SET_GIT_SAFE_DIRECTORY="${1}"
-  git config --global --add safe.directory "${1}"
-}
-unset_git_safe_directory() {
-  if [[ -z "${SET_GIT_SAFE_DIRECTORY-}" ]]
-  then
-    return
-  fi
-
-  git config --global --unset safe.directory "${1}" || git config --global --unset-all safe.directory
-  if [[ ${SET_GIT_SAFE_DIRECTORY-} == "${1}" ]]
-  then
-    unset SET_GIT_SAFE_DIRECTORY
-  fi
-}
-
 # reset Git repository
 cd "${homebrew_directory}"
-set_git_safe_directory "${homebrew_directory}"
-git reset --hard
-git checkout --force master
-git branch | grep -v '\*' | xargs -n 1 git branch --delete --force || true
-unset_git_safe_directory "${homebrew_directory}"
+# avoid writing user's global config file by making
+# "${homebrew_directory}/.gitconfig" the "global" config
+# https://git-scm.com/docs/git-config#SCOPES
+git=(env XDG_CONFIG_HOME="" HOME="${homebrew_directory}" git)
```

Comment:
How about making this a function instead? Would make the syntax a bit nicer to avoid the `git[@]` each time and can override the existing `git` so we don't need to remember it in each place.

Reply:
With the environment variables overridden for the entire script, we should be able to call the `git` executable directly.